tests/test_contracts/test_nested_preconditions_example.py

```diff
@@ -0,0 +1,80 @@
+from python_ta.contracts import check_contracts
+
+
+@check_contracts
+def my_condition1(y: int):
+    """
+    Preconditions:
+    - my_condition2(y)
+    """
+    return y > 0
+
+
+@check_contracts
+def my_condition2(x: int):
+    """
+    Preconditions:
+    - x > 0
+    """
+    return x > 0
+
+
+@check_contracts
+def my_function(z: int):
+    """
+    Preconditions:
+    - my_condition1(z)
+    """
+    return z
+
+
+def validate_student_number(student_number: int):
+    """Validate the student number of a student"""
+    return student_number > 0
+
+
+@check_contracts
+class Student:
+    """
+    A representation of a student
+
+    Representation Invariants:
+     - validate_student_number(self.student_number)
+     - self.condition2(self.gpa)
```

Comment:
Try running this type of example with `DEBUG_CONTRACTS` turned on. You'll see that this essentially triggers an infinite recursion situation (somewhat indirectly, through `eval`) because each time a method is called, the representation invariants are checked again, or something like that.

Anyway, this is a current limitation of PythonTA so I don't think you need to worry about fixing it, for now you should just avoid calling methods as representation invariants, and instead use the other top-level functions in this file.

Reply:
Ah, this makes a lot of sense! I have just noticed that I was having issues creating valid instances of the class, as no AssertionErrors were raised, and that representation invariants kept being checked cyclically, though I was not sure why.